src/librustc_trans/base.rs

```diff
@@ -1008,31 +1010,23 @@ pub fn trans_crate<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
         rx);
 
     // Translate an allocator shim, if any
-    //
-    // If LTO is enabled and we've got some previous LLVM module we translated
-    // above, then we can just translate directly into that LLVM module. If not,
-    // however, we need to create a separate module and trans into that. Note
-    // that the separate translation is critical for the standard library where
-    // the rlib's object file doesn't have allocator functions but the dylib
-    // links in an object file that has allocator functions. When we're
-    // compiling a final LTO artifact, though, there's no need to worry about
-    // this as we're not working with this dual "rlib/dylib" functionality.
-    let allocator_module = if tcx.sess.lto() {
-        None
-    } else if let Some(kind) = tcx.sess.allocator_kind.get() {
```

Comment:
Oh, that's nice that we can get rid of the special casing here. If it's easy, it would be nice to make sure we don't serialize the big "actual-code-module" and merge that into the tiny allocator module. Although it's a special case to even have an allocator module, right?

Reply:
Good point! I'll just switch LTO to merging everything into the "costliest" module